test/e2e/suite_test.go

```diff
@@ -93,11 +79,30 @@ var _ = BeforeSuite(func() {
 	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
+
+	LwsReadyForTesting(k8sClient)
 })
 
 var _ = AfterSuite(func() {
 	cancel()
-	By("tearing down the test environment")
-	err := testEnv.Stop()
-	Expect(err).NotTo(HaveOccurred())
 })
+
+func LwsReadyForTesting(client client.Client) {
+	By("waiting for ")
+
+	// To verify that webhooks are ready, let's create a simple lws.
+	lws := testutils.BuildLeaderWorkerSet(metav1.NamespaceDefault).Replica(3).Obj()
+
+	// Once the creation succeeds, that means the webhooks are ready
+	// and we can begin testing.
+	Eventually(func() error {
+		return client.Create(ctx, lws)
+	}, timeout, interval).Should(Succeed())
+
```

Comment:
could you verify that the some of the lws's spec is as expected?

Reply:
We could but it would duplicate the functionality of the actual e2e test, creating this lws is a consequence of needing to check that the webhooks are ready

Reply:
sg, then we need to add several test cases after setup the suite